cc/system_library.bzl

```diff
@@ -0,0 +1,448 @@
+
+def _find_compiler(repo_ctx):
+    gcc = _execute_bash(repo_ctx, "which g++")
+    clang = _execute_bash(repo_ctx, "which clang++")
+    if gcc:
+        return gcc
+    elif clang:
+        return clang
+    else:
+        fail("No compiler found")
+
+def _find_lib_path(repo_ctx, lib_name, archive_names, lib_path_hints):
+    override_paths = _get_list_from_env_var(
+        repo_ctx,
+        BAZEL_LIB_OVERRIDE_PATHS_ENV_VAR,
+        lib_name,
+    )
+    additional_paths = _get_list_from_env_var(
+        repo_ctx,
+        BAZEL_LIB_ADDITIONAL_PATHS_ENV_VAR,
+        lib_name,
+    )
+
+    # Directories will be searched in order
+    path_flags = _make_flags(
+        override_paths + lib_path_hints + additional_paths,
+        "-L",
+    )
+    linker = _find_linker(repo_ctx)
+    for archive_name in archive_names:
+        cmd = """
+              {} -verbose -l:{} {} 2>/dev/null | \\
+              grep succeeded | \\
+              head -1 | \\
+              sed -e 's/^\\s*attempt to open //' -e 's/ succeeded\\s*$//'
+              """.format(
+            linker,
+            archive_name,
+            path_flags,
+        )
+        path = _execute_bash(repo_ctx, cmd)
+        if path:
+            return (archive_name, path)
+    return ("", "")
+
+def _find_header_path(repo_ctx, lib_name, header_name, includes):
+    override_paths = _get_list_from_env_var(
+        repo_ctx,
+        BAZEL_INCLUDE_OVERRIDE_PATHS_ENV_VAR,
+        lib_name,
+    )
+    additional_paths = _get_list_from_env_var(
+        repo_ctx,
+        BAZEL_INCLUDE_ADDITIONAL_PATHS_ENV_VAR,
+        lib_name,
+    )
+
+    # See https://gcc.gnu.org/onlinedocs/gcc/Directory-Options.html
+    override_include_flags = _make_flags(override_paths, "-I")
+    standard_include_flags = _make_flags(includes, "-isystem")
+    additional_include_flags = _make_flags(additional_paths, "-idirafter")
+
+    compiler = _find_compiler(repo_ctx)
+
+    # Taken from https://stackoverflow.com/a/63052918/922404
+    cmd = """
+          f=\"{}\"; \\
+          echo | \\
+          {} -E {} {} {} -Wp,-v - 2>&1 | \\
+          sed '\\~^ /~!d; s/ //' | \\
+          while IFS= read -r path; \\
+              do if [[ -e \"$path/$f\" ]]; \\
+                  then echo \"$path/$f\";  \\
+                  break; \\
+              fi; \\
+          done
+          """.format(
+        header_name,
+        compiler,
+        override_include_flags,
+        standard_include_flags,
+        additional_include_flags,
+    )
+    return _execute_bash(repo_ctx, cmd)
+
+def _system_library_impl(repo_ctx):
+    repo_name = repo_ctx.attr.name
+    includes = repo_ctx.attr.includes
+    hdrs = repo_ctx.attr.hdrs
+    optional_hdrs = repo_ctx.attr.optional_hdrs
+    deps = repo_ctx.attr.deps
+    lib_path_hints = repo_ctx.attr.lib_path_hints
+    static_lib_names = repo_ctx.attr.static_lib_names
+    shared_lib_names = repo_ctx.attr.shared_lib_names
+
+    static_lib_name, static_lib_path = \
```

Comment:
Please use parenthesis instead of backslashes for line continuation here and everywhere else in the file.